src/authInfo.ts

```diff
@@ -603,6 +659,44 @@ export class AuthInfo extends AsyncCreatable<AuthInfo.Options> {
     return sfdxAuthUrl;
   }
 
+  /**
+   * Set the defaultusername or the defaultdevhubusername to the alias if
+   * it exists otherwise to the username. Method will try to set the local
+   * config first but will dfeault to global configg if that fails.
+   * @param options
+   */
+  public async setAsDefault(options: { defaultUsername?: boolean; defaultDevhubUsername?: boolean }) {
+    let config: Config;
+    // if we fail to create the local config, default to the global config
+    try {
+      config = await Config.create({ isGlobal: false });
+    } catch {
+      config = await Config.create({ isGlobal: true });
+    }
+
+    const value = ensureString(this.getFields().alias || this.getUsername());
+
+    if (options.defaultUsername) {
+      config.set(Config.DEFAULT_USERNAME, value);
+    }
+
+    if (options.defaultDevhubUsername) {
+      config.set(Config.DEFAULT_DEV_HUB_USERNAME, value);
+    }
+    await config.write();
+  }
+
+  /**
+   * Sets the provided alias to the username
+   * @param alias alias to set
+   */
+  public async setAlias(alias: string) {
+    const authFields = this.getFields();
+    await Aliases.parseAndUpdate([`${alias}=${authFields.username}`]);
+    const updated = Object.assign({}, authFields, { alias });
```

Comment:
I didn't think we stored the alias in the stored file... that seems wrong to me.

On a separate note, doesn't `this.update({ alias })` do all the merging for you?

Reply:
@amphro It exists on the AuthFields interface (https://github.com/forcedotcom/sfdx-core/pull/294/files#diff-394ba1de20ae85e42959125d8ebc62b6R50) so it _could_ be there. However, I agree that the `Aliases` class should be the single source of truth for aliases. I'll change this accordingly